packages/openapi3/test/tsp-openapi3/test-host.ts

```diff
@@ -0,0 +1,46 @@
+import { createTestWrapper } from "@typespec/compiler/testing";
+import assert from "node:assert";
+import { convertOpenAPI3Document } from "../../src/index.js";
+import { OpenAPI3Document, OpenAPI3Schema, Refable } from "../../src/types.js";
+import { createOpenAPITestHost } from "../test-host.js";
+
+export async function createConverterTestRunner() {
+  const host = await createOpenAPITestHost();
+
+  return createTestWrapper(host, {
+    wrapper(code) {
+      // Find the 1st namespace declaration and decorate it
+      const serviceIndex = code.indexOf("@service");
+      return `${code.slice(0, serviceIndex)}@test\n${code.slice(serviceIndex)}`;
+    },
+  });
+}
+
+export interface OpenAPI3Options {
+  schemas?: Record<string, Refable<OpenAPI3Schema>>;
+}
+
+export async function tspForOpenAPI3({ schemas }: OpenAPI3Options) {
+  const openApi3Doc: OpenAPI3Document = {
+    info: {
+      title: "Test Service",
+      version: "1.0.0",
+    },
+    openapi: "3.0.0",
+    paths: {},
+    components: {
+      schemas: {
+        ...(schemas as any),
+      },
+    },
+  };
+
+  const runner = await createConverterTestRunner();
+  const code = await convertOpenAPI3Document(openApi3Doc);
+  const { TestService } = await runner.compile(code);
```

Comment:
if you want could probably just use the base test host here instead of making a runner where the goal is to reuse the common pieces

Reply:
Not sure how much it saved but I switched to using the base test host.